wiring/src/spark_wiring_cellular.cpp

```diff
@@ -26,30 +26,18 @@
 namespace spark {
 
     CellularSignal CellularClass::RSSI() {
-// TODO: remove once rssi/qual are removed
-#pragma GCC diagnostic push
-#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
         CellularSignal sig;
         if (!network_ready(*this, 0, NULL)) {
-            sig.rssi = 0;
             return sig;
         }
 
-        CellularSignalHal sig_hal = {0};
         cellular_signal_t sigext = {0};
         sigext.size = sizeof(sigext);
-        if (cellular_signal(&sig_hal, &sigext) != 0) {
-            sig.rssi = 1;
+        if (cellular_signal(nullptr, &sigext) != 0) {
             return sig;
         }
-        sig.rssi = sig_hal.rssi;
-        sig.qual = sig_hal.qual;
-        if (sig.rssi == 0) {
-            sig.rssi = 2;
-        }
```

Comment:
I want to make sure we don't get out of sync with our documentation here.  sig.rssi = 2 and sig.rssi = 1 are special error cases.  If we are deprecating those errors as well, we should be sure to update the docs.  Otherwise perhaps we need to keep the errors?  I can't seem to find anything about Cellular.RSSI() errors in our Docs anymore... but that could be it's own issue.  So I'll just ask this question, if the API to request signal strength/quality fails, how does the user know there is an error?

Reply:
Part of documentation already says this but will need to update docs as needed. 
```
Before Device OS 0.8.0, the CellularSignal class only had two member variables, rssi, and qual. These will be removed in a future version of Device OS and you should use getStrengthValue() and getQualityValue() instead.
``` 

I don't think it's a good idea to keep these errors as it could confuse the users even more. Currently, when the signal strength/quality fails we should get `0` on the `getStrengthValue` and `getQualityValue` and a `-1` on `getStrength` and `getQuality`.

Reply:
Good point, Brett. I suggest we add `operator bool` to `CellularSignal`.

Reply:
The problem is that we need to decide on the behavior of that operator:

1. RAT is unknown - a clear failure
2. RAT is valid, strength/quality are invalid/unavailable (temporarily unavailable as modem needs some time to calculate them for example) - ?
3. RAT is valid, strength is valid, quality is invalid/unavailable (isn't supported on some modems with certain RATs for example, or again, the modem needs some time to calculate it) - ?

Reply:
I think it's very possible to have a RAT, but have a missing RSRP/RSRQ value... which results in 0 or -1 in the various APIs.  If we can only use an operator bool like we do in other APIs, perhaps if any error is present this returns false.  This would exclude when quality is not supported.  If we'd like to know exactly what element failed or is not valid, perhaps the user could then turn to individual tests on each one (!= -1 or != 0)... and we'd need to make sure these are all documented (currently I don't see these documented).

Reply:
Yeah, IMO, that operator should return `true` if the object as a whole is valid, i.e. contains RAT, signal strength and quality, if the latter is applicable. Alternatively, instead of the operator, we can have separate methods like `isRatValid()`, `isStrengthValid()`, etc.

Reply:
For consistency, this should probably be `operator bool()` + `isValid()` as with other wiring APIs. Forwarding info about unsupported quality from HAL should be doable as `cellular_signal_t` is extendable.

Reply:
Then, perhaps, let's give the users full visibility into the info available:

`isAccessTechnologyValid()`
`isStrengthValid()`
`isQualityValid()`
`isQualitySupported()`

?

Reply:
Similar to rssi error codes before, If we can provide one API which returns an `int` and the users can check a look up table?
- `0` - No error
- `1` - Invalid RAT
- `2` - Invalid Strength but valid quality
- `3` - Valid strength but invalid quality
- `4` - InValid strength and quality

Reply:
Distinct named methods are the way if we are to add granular checks, because at the end of the day what we are interested in is whether a particular piece of info is valid/available.

Reply:
Sounds like an agreement on the bool operator.  I'm wondering what the cost is of adding all of the distinct methods for validity checks?  Once we add them, it would take a while to deprecate them if need be for space or something.  Should we also add them at this time, and/or just document the validity check for now?

Reply:
Let's also ask @jlovinger-particle as he probably has a better perspective as to which approach would be more convenient/useful from the UX point of view. I'd be fine with any approach as long as we're not forcing our users to deal with special values – API deprecation is a good opportunity to make improvements like this.